src/components/Footer.tsx

```diff
@@ -0,0 +1,82 @@
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { Filter } from '../types/Filter';
+
+type Props = {
+  filter: Filter,
+  onSetFilter: (filter: Filter) => void,
+  completedTodos: Todo[],
+  onRemoveAll: (userId: number) => void,
+  activeTodos: Todo[];
+};
+
+export const Footer: React.FC<Props> = ({
+  filter,
+  onSetFilter,
+  completedTodos,
+  onRemoveAll,
+  activeTodos,
+}) => {
+  const removeAllcompleted = () => {completedTodos.forEach(todo => onRemoveAll(todo.id))};
+  const handlerFilterAll = () => onSetFilter(Filter.All);
+  const handlerFilterActive = () => onSetFilter(Filter.Active);
+  const handlerFilterCompleted = () => onSetFilter(Filter.Completed);
```

Comment:
Instead of creating 3 handlers you can do one
```
const onFilterChange = (filter: Filter) => () => {
  onSetFilter(filter)
}
```
and usage 
```
onClick={onFilterChange(Filter.All)}
...
onClick={onFilterChange(Filter.Active)}
...
onClick={onFilterChange(Filter.Completed)}
```